lib/pipeline/RequestHandler.js

```diff
@@ -115,18 +123,15 @@ class RequestHandler extends Duplex {
   }
 
   _flush() {
-    var i;
-
-    //faster than deque.
-    for(i = 0; i < this.queue.length && this.reading ; i++) {
-      this.reading = this.push(this.queue[i]);
+    while (this.queue.length) {
+      var queued = this.queue.shift();
```

Comment:
If `queue` is an array using `shift` might have a perf penalty because it has to move all elements one position left. Also it is not recommended to modify a collection you are iterating over.